server/rest/src/main/java/org/hawkular/apm/server/rest/ServiceDeploymentHandler.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2015-2017 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.apm.server.rest;
+
+import static javax.ws.rs.core.MediaType.APPLICATION_JSON;
+import static javax.ws.rs.core.MediaType.APPLICATION_JSON_TYPE;
+
+import java.util.List;
+
+import javax.inject.Inject;
+import javax.ws.rs.BeanParam;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.Response;
+
+import org.hawkular.apm.api.model.analytics.ServiceDeployment;
+import org.hawkular.apm.api.services.ServiceDeploymentService;
+import org.hawkular.apm.server.rest.entity.TenantRequest;
+import org.jboss.logging.Logger;
+
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+
+/**
+ * JAX-RS handler for Service Deployments. A service deployment is a special tag, that allows to
+ * distinguish between deployments within a specific service. For now, a service deployment is equivalent to a
+ * buildStamp, which follows the pattern "{service}-{serialNumber}". In a future iteration, it might mean a set of
+ * specific buildStamps acting together among services.
+ *
+ * @author Juraci Paixão Kröhling
+ */
+@Path("services")
+@Produces(APPLICATION_JSON)
+@Api(value = "services", description = "Service Deployments")
+public class ServiceDeploymentHandler extends BaseHandler {
+    private static final Logger log = Logger.getLogger(ServiceDeploymentHandler.class);
+
+    @Inject
+    ServiceDeploymentService serviceDeploymentService;
+
+    /**
+     * Returns the available services. Service here, at this point, are individual services. A service has 1+ buildStamps.
+     * The list of buildStamps within the service is <b>not</b> comprehensive, but limited to the 5 most recently created stamps.
+     * @return a {@link List} of {@link ServiceDeployment}, wrapped into a {@link Response}.
+     */
+    @GET
+    @Path("/")
+    @ApiOperation(value = "Returns all the available services", response = List.class)
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "Success"),
+            @ApiResponse(code = 500, message = "Internal server error") })
+    public Response getServices(@BeanParam TenantRequest request) {
```

Comment:
Criteria is not being passed in via REST API - same for single service operation below.